src/url-loader/fs-url-loader.ts

```diff
@@ -30,15 +30,13 @@ export class FSUrlLoader implements UrlLoader {
   root: string;
 
   constructor(root: string = '') {
-    if (root.endsWith('/')) {
-      root += '/';
-    }
     this.root = pathlib.resolve(root);
   }
 
   canLoad(url: ResolvedUrl): boolean {
-    return url.startsWith('file://') &&
-        isPathInside(this.root, Uri.parse(url).fsPath);
+    const parsed = Uri.parse(url);
+    return parsed.scheme === 'file' && !parsed.authority &&
```

Comment:
Previously this was testing using `startsWith`, so if there was a host/authority (which we have test for), there would be a missing slash, and it worked out. Now we need to explicitly check there is no authority.